src/dtx/dtx_common.c

```diff
@@ -300,15 +297,16 @@ dtx_aggregate(void *arg)
 		if (stat.dtx_cont_cmt_count == 0 ||
 		    stat.dtx_first_cmt_blob_time_lo == 0 ||
 		    (stat.dtx_cont_cmt_count <= dtx_agg_thd_cnt_lo &&
-		     dtx_hlc_age2sec(stat.dtx_first_cmt_blob_time_lo) <=
-		     dtx_agg_thd_age_lo))
+		     dtx_hlc_age2sec(stat.dtx_first_cmt_blob_time_lo) <= dtx_agg_thd_age_lo))
 			break;
 	}
 
 	sched_req_put(dbca->dbca_agg_req);
 	dbca->dbca_agg_req = NULL;
 
 out:
+	D_ASSERT(dbca->dbca_pool->dbpa_aggregating != 0);
+	dbca->dbca_pool->dbpa_aggregating--;
```

Comment:
I don't quite follow this, this ULT is tracked by 'dbca_agg_req', how can multiple ULTs co-exists? (I see you changed dbpa_aggregating from bool to an integer).

Reply:
Each container can have one ULT to do DTX aggregation on the container. The pool may contain multiple containers.

Reply:
oh, I see. dbca_agg_req is per vos cont but dpca_aggregating is per vos pool. Thanks.